tool/tbot/tshwrap/wrap_test.go

```diff
@@ -0,0 +1,191 @@
+type mockRunner struct {
+	captureErr error
+	captureOut []byte
+}
+
+func (r *mockRunner) Capture(args ...string) ([]byte, error) {
+	if r.captureErr != nil {
+		return nil, r.captureErr
+	}
+	return r.captureOut, nil
+}
+
+func (r *mockRunner) Exec(env map[string]string, args ...string) error {
+	return trace.NotImplemented("not implemented")
+}
+
+// TestTSHSupported ensures that the tsh version check works as expected (and,
+// implicitly, that the version capture and parsing works.)
+func TestTSHSupported(t *testing.T) {
+	version := func(v string) []byte {
+		bytes, err := json.Marshal(struct {
+			Version string `json:"version"`
+		}{
+			Version: v,
+		})
+		require.NoError(t, err)
+
+		return bytes
+	}
+
+	tests := []struct {
+		name   string
+		out    []byte
+		err    error
+		expect func(t *testing.T, err error)
+	}{
+		{
+			// Before `-f json` is supported
+			name: "very old tsh",
+			err:  trace.Errorf("unsupported"),
+			expect: func(t *testing.T, err error) {
+				require.Error(t, err)
+			},
+		},
+		{
+			name: "too old",
+			out:  version("9.2.0"),
+			expect: func(t *testing.T, err error) {
+				require.Error(t, err)
+			},
+		},
+		{
+			name: "supported",
+			out:  version(TSHMinVersion),
+			expect: func(t *testing.T, err error) {
+				require.NoError(t, err)
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			runner := &mockRunner{
+				captureOut: tt.out,
+				captureErr: tt.err,
+			}
+
+			tt.expect(t, CheckTSHSupported(runner))
+		})
+	}
+}
+
+// TestGetEnvForTSH ensures we generate a valid minimum subset of environment
+// parameters needed for tsh wrappers to work.
+func TestGetEnvForTSH(t *testing.T) {
+	destination := config.DestinationConfig{
+		DestinationMixin: config.DestinationMixin{
+			Directory: &config.DestinationDirectory{
+				Path: "/foo",
+			},
+		},
+	}
+	require.NoError(t, destination.CheckAndSetDefaults())
+
+	p, err := GetDestinationPath(&destination)
+	require.NoError(t, err)
+
+	tlsCAs, err := GetTLSCATemplate(&destination)
+	require.NoError(t, err)
+
+	expected := map[string]string{
+		client.VirtualPathEnvName(client.VirtualPathKey, nil):      filepath.Join(p, identity.PrivateKeyKey),
+		client.VirtualPathEnvName(client.VirtualPathDatabase, nil): filepath.Join(p, identity.TLSCertKey),
+		client.VirtualPathEnvName(client.VirtualPathApp, nil):      filepath.Join(p, identity.TLSCertKey),
+
+		client.VirtualPathEnvName(client.VirtualPathCA, client.VirtualPathCAParams(types.UserCA)):     filepath.Join(p, tlsCAs.UserCAPath),
+		client.VirtualPathEnvName(client.VirtualPathCA, client.VirtualPathCAParams(types.HostCA)):     filepath.Join(p, tlsCAs.HostCAPath),
+		client.VirtualPathEnvName(client.VirtualPathCA, client.VirtualPathCAParams(types.DatabaseCA)): filepath.Join(p, tlsCAs.DatabaseCAPath),
+	}
+
+	env, err := GetEnvForTSH(&destination)
+	require.NoError(t, err)
+	for k, v := range expected {
+		require.Contains(t, env, k)
+		require.Equal(t, v, env[k])
```

Comment:
Simplify? Also let all errors accumulate?

```suggestion
		assert.Equal(t, v, env[k])
```